mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -319,8 +321,8 @@ public void preStart() throws Exception {
     @Override
     public void postStop() throws Exception {
         LOGGER.info("Job Actor {} stopped invoking cleanup logic", jobId);
-        if (jobId != null) {
-            MetricsRegistry.getInstance().remove(getMetricGroupId(jobId.getId()));
+        if (jobId != null && mantisJobMetaData != null) {
+            MetricsRegistry.getInstance().remove(getMetricGroupId(jobId.getId(), getResourceCluster()));
```

Comment:
Can you not use metricsGroupId directly?

Reply:
Good point. I don't know why it wasn't done before 🤔

Reply:
updated.